### DISCUSSION TITLE
Support for new camera v3 with autofocus

### DISCUSSION THREAD

**Discussion by waisaacs** [raspberrypi/picamera2#508] [General]
Just saw https://www.raspberrypi.com/news/new-autofocus-camera-modules/ announced and wondered what it implies for picamera2 development to support it.  Will the autofocus feature make it in, eventually I suppose?  I'm keen to order one to play with.  I'll admit I haven't poked around at all for RPi support documents about it yet.

**Comment by tvoverbeek** [comment]
It is already in the next branch: https://github.com/raspberrypi/picamera2/commit/9c44552836aa721b730fafd904923b39ddb86201
So it should be in the next release.

**Reply by waisaacs** [reply]
Thanks for the info and great to hear!  I'll see how soon I can get my hands on one.

**Comment by waisaacs** [comment]
And if I would have finished reading the post I quoted before I ran here to excitedly ask about it, I would have seen the answer to my question:

The new Camera Module 3 is only supported by the modern libcamera software environment and by the libcamera-based Picamera2 beta under Raspberry Pi OS Bullseye, and not by the legacy closed-source camera stack – you’ll need to make sure you have the latest version of the software before you dig in, as only the latest release has autofocus support. But don’t worry, the hardware and software documentation has been updated to get you going, including the documentation around our Picamera2 library.
